frontend/src/components/common/SimpleTable.tsx

```diff
@@ -104,6 +163,64 @@ export default function SimpleTable(props: SimpleTableProps) {
   // eslint-disable-next-line
   [data, currentData]);
 
+  function defaultSortingFunction(column : SimpleTableGetterColumn) {
+    function defaultSortingReal(item1: any, item2: any) {
+      const value1 = column.getter(item1);
+      const value2 = column.getter(item2);
+
+      let compareValue = 0;
+      if (value1 < value2) {
+        compareValue = -1;
+      } else if (value1 > value2) {
+        compareValue = 1;
+      }
+
+      return compareValue * (isIncreasingOrder ? 1 : -1);
+    }
+
+    return defaultSortingReal;
+  }
+
+  React.useEffect(() => {
+    if (!data) {
+      return;
+    }
+    let applySort = undefined;
+    const columnAskingForSort = columns[sortColIndex];
+    const sortFunction = columnAskingForSort?.sort;
+
+    if (typeof sortFunction === 'boolean' && sortFunction){
+      setCurrentData([...data.sort(defaultSortingFunction(columnAskingForSort as
+        SimpleTableGetterColumn))]);
+      return;
+    }
+
+    if (typeof sortFunction === 'function') {
+      applySort = (arg1: any, arg2: any) => {
+        const orderChanger = isIncreasingOrder ? 1 : -1;
+        return sortFunction(arg1, arg2) as any * orderChanger;
+      };
+      const sortedData = data.sort(applySort);
+      setCurrentData([...sortedData]);
```

Comment:
AFAIU, this will set the `currentData` different from the `data`, so as soon as we change the sorting order we will get the `Refresh` button shown. Same if the same data is set up in the table when the current data is the same but sorted.
This is the reason why we use `filteredData` as a separate variable rather than filtering `currentData`.

I am not sure if sorting the `currentData` before the `filteredData` (i.e. not storing sorted data) is going to impact the performance a lot, but if it does, then maybe the solution is to either set the `data` as `currentData` on a sorting request when it's still equal to `currentData`  (so you don't miss the new data if it changed), or to use a third state variable called `displayData` (for holding the filtered/sorted data, but this seems to create confusion as well).

Reply:
yes you are right, but data is a prop to the SimpleTable component and we should never modify prop directly in a component, Thus we are only left with the confusing solution which is to use one more state called displayData